sound/soc/sof/intel/hda-dai.c

```diff
@@ -33,6 +33,43 @@ struct hda_pipe_params {
 	unsigned int link_bps;
 };
 
+int hda_dai_config(struct snd_sof_dev *sdev,
+					struct snd_soc_dai_link *link,
+					struct sof_ipc_dai_config *config,
+					int direction)
+{
+	struct snd_pcm_substream substream;
+	struct snd_soc_dai *dai;
+	struct hdac_ext_stream *stream;
+	struct snd_soc_dai_link_component dai_component = {0};
+	struct hdac_bus *bus = sof_to_bus(sdev);
```

Comment:
init needed?
Also reverse xmas tree order.

Reply:
yes, I will refine it.